### PR TITLE
improved structure and readability of main.py

### PR DESCRIPTION
**Pull request by AnastasiosZyngiridis**
1.Removed unnecessary import statements (import sys, import shutil, import getpass).

2.Encapsulated the code inside the BotCore class to provide better organization and encapsulation.

3.Replaced hardcoded configuration values with variables (self.server_url, self.port, self.name, etc.).

4.Utilized the os.path module for path-related operations instead of manually concatenating strings.

5.Updated the logging statements to use the logger instance.

6.Refactored the message_info method to handle exceptions and return a dictionary with all fields set to prevent None values.

7.Updated the methods method to return a dictionary of method references.

8.Added docstrings to functions and methods.

9.Fixed the indentation of the code to conform to Python style guidelines.

10.Replaced the print statements with logger statements for consistent logging.

11.Modified the load_plugins_from_folder method to handle different sources of plugin names (from configuration file or directory) and log errors appropriately.

12.Updated the load_plugins method to load plugins from the specified folders.

13.Refactored the run_plugins method to use the loaded plugins and pass the necessary arguments to the plugin's run method.

14.Renamed command_parser method to core_commands_parse for clarity.

15.Removed unused methods and variables (self.sp_command, self.core_plugins, pull).

16.Added missing exception handling in the quit method.

17.Cleaned up the code formatting and removed unnecessary comments.

## Additional context

Add any other context or screenshots about the pull request here.

## Checklist

- [x] I have read, understood and followed our [Contributing Guide](https://github.com/pyhoneybot/honeybot/blob/master/CONTRIBUTING.md)
- [ ] My code follows the code style of this project
- [ ] I have updated the documentation accordingly
- [ ] I added my country flag to the `README.md`

## Code of Conduct

By submitting this issue, you agree to follow our [Code of Conduct](https://github.com/pyhoneybot/honeybot/blob/master/CODE_OF_CONDUCT.md)

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

File: src/honeybot/api/main.py [hunk 1]
```diff
@@ -245,49 +241,24 @@ def greet(self):
...
+
+            except KeyboardInterrupt:
+                self.is_listen_on = False
+                self.quit()
             except Exception as e:
-                logger.info(e)
-
-    """
-    ONGOING REQUIREMENT/S
-    """
```
Comment (Abdur-rahmaanJ) [inline src/honeybot/api/main.py:268]: @AnastasiosZyngiridis Why did you remove this section?

Comment (AnastasiosZyngiridis) [reply src/honeybot/api/main.py:268]: @Abdur-rahmaanJ Sorry for the mistake. As they where ONGOING REQUIREMENT/S i didn't touch them and forgot to put them back in after i was done editing the rest of the code.

**Review by Abdur-rahmaanJ — CHANGES_REQUESTED** [review]
Put code back

**Review by Abdur-rahmaanJ — APPROVED** [review]
l
